src/ucp/wireup/address.c

```diff
@@ -951,7 +951,8 @@ ucs_status_t ucp_address_pack(ucp_worker_h worker, ucp_ep_h ep,
 
 ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
                                 unsigned unpack_flags,
-                                ucp_unpacked_address_t *unpacked_address)
+                                ucp_unpacked_address_t *unpacked_address,
+                                size_t *size_p)
```

Comment:
maybe return ssize_t instead of returning size_p by value?

Reply:
But this method returns `ucs_status_t`

Reply:
yes, return ssize_t instead of ucs_status_t

Reply:
But there are cases when there could be errors in address: https://github.com/openucx/ucx/blob/master/src/ucp/wireup/address.c#L1154

Reply:
right, return the code. ssize_t is signed type, can be negative err or positive size

Reply:
Would need to change a lot of code where it used like this:
```
   status = ucp_address_unpack(worker, ucp_addr, addr_pack_flags,
                                &unpacked_addr);
    if (status != UCS_OK) {
        goto free_ucp_addr;
    }
```

Reply:
As discussed  to be compatible with ucp_address_pack, that uses pointer argument.